python/lsst/validate/drp/checkAstrometryPhotometry.py

```diff
@@ -0,0 +1,514 @@
+    matchRef : int, optional
+        Should match at least matchRef stars.
+
+    Returns
+    -------
+    float
+        The photometry scatter (RMS, millimag) for all good stars.
+
+    Notes
+    -----
+    The scatter and match defaults are appropriate to SDSS are stored here.
+    For SDSS, stars with mag < 19.5 should be completely well measured.
+    This limit is a band-dependent statement most appropriate to r.
+    """
+
+    print("Median value of the photometric scatter - all magnitudes: %.3f %s" %
+          (np.median(mmagrms), "mmag"))
+
+    bright = np.where(np.asarray(mag) < good_mag_limit)
+    photoScatter = np.median(np.asarray(mmagrms)[bright])
+    print("Photometric scatter (median) - mag < %.1f : %.1f %s" %
+          (good_mag_limit, photoScatter, "mmag"))
+
+    if photoScatter > medianRef:
+        print("Median photometric scatter %.3f %s is larger than reference : %.3f %s " % (photoScatter, "mmag", medianRef, "mag"))
+    if match < matchRef:
+        print("Number of matched sources %d is too small (shoud be > %d)" % (match, matchRef))
+
+    return photoScatter
+
+
+def printPA2(gv, magKey):
+    """Calculate and print the calculated PA2 from the LSST SRD from a groupView."""
+
+    pa2 = calcPA2(gv, magKey)
+    print("minimum: PF1=%2d%% of diffs more than PA2 = %4.2f mmag (target is PA2 < 15 mmag)" % 
+          (pa2.PF1['minimum'], pa2.minimum))
+    print("design:  PF1=%2d%% of diffs more than PA2 = %4.2f mmag (target is PA2 < 15 mmag)" % 
+          (pa2.PF1['design'], pa2.design))
+    print("stretch: PF1=%2d%% of diffs more than PA2 = %4.2f mmag (target is PA2 < 10 mmag)" % 
+          (pa2.PF1['stretch'], pa2.stretch))
+
+def repoNameToPrefix(repo):
+    """Generate a base prefix for plots based on the repo name.
+
+    Examples
+    --------
+    >>> repoNameToPrefix("a/b/c")
+    "a_b_c"
+    >>> repoNameToPrefix("/bar/foo/")
+    "bar_foo"
+    >>> repoNameToPrefix("CFHT/output")
+    "CFHT_output"
+    """
+
+    dirnames = []
+    remaining_path = repo
+    while os.path.split(remaining_path):
+        remaining_path, tail = os.path.split(remaining_path)
+        dirnames.append(tail)
+        if remaining_path in ["", "/"]: 
+            break
+
+    base = "_".join(dirnames[::-1])  # reverse to get order right
+    return base
+
+####
+def run(repo, visitDataIds, good_mag_limit, 
+        medianAstromscatterRef=25, medianPhotoscatterRef=25, matchRef=500):
+    """Main executable.
+
+    Inputs
+    ------
+    repo : string
+        The repository.  This is generally the directory on disk
+        that contains the repository and mapper.
+    visitDataIds : list of dict
+        List of `butler` data IDs of Image catalogs to compare to reference.
+        The `calexp` cpixel image is needed for the photometric calibration.
+    good_mag_limit : float
+        Minimum average brightness (in magnitudes) for a star to be considered.
+    """
+
+    plotbase = repoNameToPrefix(repo)
```

Comment:
Ah, good catch!  I didn't think about this.  I knew I wasn't completely happy with `repoNametoPrefix`, and still am not.  For now I'll explicitly handle this './' very common case.  I've added to doctest examples to define this.
```
    >>> repoNameToPrefix('./CFHT/output')
    'CFHT_output_'
    >>> repoNameToPrefix('.a/CFHT/output')
    'a_CFHT_output_'
```

Note this isn't quite consistent in some larger sense, but I'm fine with the code failing to produce distinct names for '.a/' and 'a/'.  The purpose was to generate different filenames for now.  Eventually this will all be hooked into the test harness as part of DM-2050.